pkg/cmd/pulumi/new.go

```diff
@@ -390,6 +451,205 @@ func useSpecifiedDir(dir string) (string, error) {
+		opts,
+		args.aiLanguage,
+		args.aiPrompt,
+		args.yes,
+		userName,
+		"",
+		"",
+		"",
+		"",
+	); continuePrompt == RefineSelection; continuePrompt,
+		conversationURL,
+		conversationID,
+		connectionID,
+		err = runAINewPromptStep(
+		opts,
+		args.aiLanguage,
+		args.aiPrompt,
+		args.yes,
+		userName,
+		continuePrompt,
+		connectionID,
+		conversationURL,
+		conversationID,
+	) {
+		if err != nil {
+			return "", err
+		}
+	}
+	return conversationURL, err
+}
+
+func sendPromptToPulumiAI(
+	promptMessage string,
+	conversationID string,
+	connectionID string,
+	userName string,
+	language string,
+) (string, string, string, error) {
+	pulumiAIURL := env.AIServiceEndpoint.Value()
+	if pulumiAIURL == "" {
+		pulumiAIURL = "https://www.pulumi.com/ai"
+	}
+	parsedURL, err := url.Parse(pulumiAIURL)
+	if err != nil {
+		return "", "", "", err
+	}
+	requestPath := parsedURL.JoinPath("api", "chat")
+	requestBody := AIPromptRequestBody{
+		Language:       language,
+		Instructions:   promptMessage,
+		Model:          "gpt-4-turbo",
+		ResponseMode:   "code",
+		ConversationID: conversationID,
+		ConnectionID:   connectionID,
+	}
+	marshalledBody, err := json.Marshal(requestBody)
+	if err != nil {
+		return "", "", "", err
+	}
+	userDataCookie := http.Cookie{Name: "pulumi_command_line_user_name", Value: userName}
+	request, err := http.NewRequest("POST", requestPath.String(), bytes.NewReader(marshalledBody))
+	if err != nil {
+		return "", "", "", err
+	}
+	request.AddCookie(&userDataCookie)
+	fmt.Println("Sending prompt to Pulumi AI...")
+	res, err := http.DefaultClient.Do(request)
+	if err != nil {
+		return "", "", "", err
+	}
+	defer res.Body.Close()
+	reader := bufio.NewReader(res.Body)
+	fmt.Println("Pulumi AI response:")
+	for {
+		chunk, err := reader.ReadByte()
+		if err != nil && err.Error() != "EOF" {
+			return "", "", "", err
+		}
+		fmt.Print(string(chunk))
+		if err != nil && err.Error() == "EOF" {
+			// Add a newline to ensure we don't overwrite the last line of the Pulumi AI response
+			fmt.Println()
+			break
+		}
+	}
+	conversationID = res.Header.Get("x-conversation-id")
+	connectionID = res.Header.Get("x-connection-id")
+	conversationURL := parsedURL.JoinPath("api", "project", url.PathEscape(fmt.Sprintf("%s.zip", conversationID))).String()
+	return conversationURL, connectionID, conversationID, nil
+}
+
+func runAINewPromptStep(
+	opts display.Options,
+	language string,
+	prompt string,
+	yes bool,
+	userName string,
+	currentContinueSelection string,
+	connectionID string,
+	conversationURL string,
+	conversationID string,
+) (continueSelection string,
+	conversationURLReturn string,
+	conversationIDReturn string,
+	connectionIDReturn string,
+	err error,
+) {
+	var promptMessage string
+	if prompt == "" || currentContinueSelection != "" {
+		if err := survey.AskOne(&survey.Input{
+			Message: "Please input your prompt here:\n",
```

Comment:
Perhaps offer a suggestion like in pulumi.com/ai to avoid "blank screen" problems?

```suggestion
			Message: "Please input your prompt here ("a static website on AWS behind a CDN"):\n",
```